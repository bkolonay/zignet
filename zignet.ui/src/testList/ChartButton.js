import React, { Component } from 'react';
import { getUrlParams } from '../common/routing/SearchFilter.js'

class ChartButton extends Component {

  render() {
    return (
      <a href={"/?" + getUrlParams(this.props.filter)} className="btn btn-primary float-right" role="button" title="View results as charts">Chart</a>
    );
  }
}

export default ChartButton;
