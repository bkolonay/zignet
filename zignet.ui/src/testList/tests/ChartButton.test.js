import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChartButton from '../ChartButton';

it('renders', () => {
  const component = <ChartButton />;
  ReactDOM.render(component,document.createElement('div'));
});

it('snapshot', () => {
  const component = <ChartButton />;
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});