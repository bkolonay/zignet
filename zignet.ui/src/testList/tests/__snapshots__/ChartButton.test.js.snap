// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<a
  className="btn btn-primary float-right"
  href="/"
  role="button"
  title="View results as charts"
>
  Chart
</a>
`;
